src/android/com/adobe/phonegap/push/GCMIntentService.java

```diff
@@ -87,8 +87,12 @@ protected void onMessage(Context context, Intent intent) {
         // Extract the payload from the message
         Bundle extras = intent.getExtras();
         if (extras != null) {
+
+            SharedPreferences prefs = context.getSharedPreferences(PushPlugin.COM_ADOBE_PHONEGAP_PUSH, Context.MODE_PRIVATE);
+            boolean forceShow = prefs.getBoolean(FORCE_SHOW, false);
+
             // if we are in the foreground, just surface the payload, else post it to the statusbar
-            if (PushPlugin.isInForeground()) {
+            if (!forceShow && PushPlugin.isInForeground()) {
```

Comment:
Since we have two variables we need to account for 4 different cases. 

1) forceShow = false & PushPlugin.isInForeground() = true
- should set foreground boolean to true and then fire the `notification` event.

2) forceShow = true & PushPlugin.isInForeground() = true
- should set foreground boolean to true, fire the `notification` event and show the notification in the shade

3) forceShow = false & PushPlugin.isInForeground() = false
- should set foreground boolean to false and then show the notification in the shade

4) forceShow = true & PushPlugin.isInForeground() = false
- should set foreground boolean to false and then show the notification in the shade

Obviously 3 and 4 are the same so we can handle that in an else but we need something like this un-optimized code:

```
            if (!forceShow && PushPlugin.isInForeground()) {
                extras.putBoolean(FOREGROUND, true);
                PushPlugin.sendExtras(extras);
            } else if (forceShow && PushPlugin.isInForeground()) {
                extras.putBoolean(FOREGROUND, true);
                PushPlugin.sendExtras(extras);

                // Send a notification if there is a message
                String message = this.getMessageText(extras);
                String title = getString(extras, TITLE, "");
                if ((message != null && message.length() != 0) ||
                        (title != null && title.length() != 0)) {
                    createNotification(context, extras);
                }
            } else {
                extras.putBoolean(FOREGROUND, false);

                // Send a notification if there is a message
                String message = this.getMessageText(extras);
                String title = getString(extras, TITLE, "");
                if ((message != null && message.length() != 0) ||
                        (title != null && title.length() != 0)) {
                    createNotification(context, extras);
                }
            }
```